docs/articles/documentation/why-testcafe/README.md

```diff
@@ -0,0 +1,126 @@
+* TestCafe tests are Node.js scripts. They can launch services and applications, read and write system files, make use of your favorite libraries.
+* TestCafe uses client-side automation scripts to execute in-browser actions. This is how our testing library handles asynchronous events, simulates user activity, and executes user-defined JavaScript.
+
+Early versions of TestCafe ran entirely in the browser. Moving to a hybrid architecture allowed us to improve the tests' stability and expand the framework's range of testing capabilities. 
+
+### Page Proxying: Bridging the Client-Server Gap
+
+You may notice that when you run TestCafe, the URL in the browser’s address bar does not match that of your website. This happens because TestCafe runs an under-the-hood reverse proxy.
+
+The [testcafe-hammerhead](https://github.com/DevExpress/testcafe-hammerhead) proxy intercepts browser requests and injects automation scripts into the requested pages. When the proxy receives data, it changes all the URLs on the resource so that they point to the proxy. This means that neither the client-side code nor other resources in use can tell that the page has been modified. To conceal automation scripts from the rest of the page code, TestCafe also intercepts some of the requests to the browser API.
+
+The proxying mechanism ensures that the page appears to be hosted at the original URL even to the test code.
+
+### Browser Sandboxing
+
+At the end of each run, TestCafe deletes all browser cookies, empties the storage, and reloads the page, thereby preventing undesirable interference with subsequent tests. You don’t need to write boilerplate code to reset the app state and reverse the changes your tests make.
+
+Tests that run in parallel operate in independent sandboxed environments. This helps prevent server-side collisions.
+
+### Client-Side Scripts
+
+TestCafe translates server-side test code into client-side JavaScript, and injects it into the browsers that it controls. This process enables the framework to perform common in-browser actions. Some testing scenarios, however, require the execution of custom client-side code. There are three ways to do it with TestCafe:
+
+[Client Scripts](/../guides/advanced-guides/inject-client-scripts.md) inject custom JavaScript files, such as temporary extra dependencies, into the page.
+[Client Functions](../guides/basic-guides/obtain-client-side-info.md) evaluate user-defined JavaScript expressions and pass their return value to the server side. They are useful when you want to examine the page or access its URL.
+The [Selector](../guides/basic-guides/select-page-elements.md) function can launch user-defined client-side code to find a DOM element that cannot be otherwise identified.
+
+> The TestCafe documentation describes the limitations of user-defined client-side scripts.
```

Comment:
It's a bit weird to end the article like that... We talk about limitations - being negative. 

Maybe we should add a section that refers people to getting started resources? The message in the end should be something like "ok, you like what you just read? Then let's go!" :-)